### PR TITLE
adding debug logging to help with diagnostics

### PR DESCRIPTION
**Pull request by PiotrMag7**
This is a work to address refresh token issue, where on occasion refresh token gets deleted from hydra db.
This happens during refresh token flow. The flow includes series of actions: delete access tokens from sql, delete access tokens from casssandra, delete refresh token from sql, delete refresh token from cassandra, create or insert the same refresh token to sql and than cassandra (the same if the token is configured with refreshtime = -1 , which means never expire), create/insert new access token both sql and cassandra.
These operations are not within transaction so when something fails before token is re-inserted the token is gone, meaning session is gone and api user can't anymore refresh her token.
To mitigate we:
- implemented retries
- change order of operations - performing cassandra first (which we think is causing issues). This way if the operation fails it will make cassandra out of the loop, but we think its less disruptive.
Hopefully re-tries will remedy situation.

### CODE REVIEW

File: access_request_handler.go [hunk 1]
```diff
@@ -102,14 +105,15 @@ func (f *Fosite) NewAccessRequest(ctx context.Context, r *http.Request, session
 		if err := loader.HandleTokenEndpointRequest(ctx, accessRequest); err == nil {
 			found = true
 		} else if errors.Cause(err).Error() == ErrUnknownRequest.Error() {
+			//TODO: some logging, requires instrumentation
```
Comment (paddyspotx) [inline access_request_handler.go:108]: do you want to implement this instrumentation in this change?

File: errors.go [hunk 2]
```diff
@@ -249,6 +249,7 @@ func (e *RFC6749Error) StatusCode() int {
 
 func (e *RFC6749Error) WithDebug(debug string) *RFC6749Error {
 	err := *e
-	err.Debug = debug
+	//TODO - PK: check if this handle empty initial Debug
```
Comment (paddyspotx) [inline errors.go]: A TODO still here.

**Review by paddyspotx — DISMISSED** [review]
Just some highlighting of comments/todos left in. otherwise lgtm.

File: access_request_handler.go (hunk 1 above)
Comment (PiotrMag7) [reply access_request_handler.go:108]: no

**Review by pmccall-magnite — APPROVED** [review]

**Review by pmccall-magnite — APPROVED** [review]

File: access_request_handler.go (hunk 1 above)
Comment (mgloystein) [reply access_request_handler.go:108]: Good plan...

**Review by jscullion — APPROVED** [review]
LGTM

**Pull request merged**